File: Sliding_Window/LC_121.py
# Written by: Jian-Yuan Lin
# Date: 2022.11.28

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        prev_min, ans = prices[0], 0
        for i in range(1, len(prices)):
            price = prices[i]
            ans = max(price - prev_min, ans)
            prev_min = min(price, prev_min)
        return ans

File: Sliding_Window/LC_121.cpp
// Written by: Jian-Yuan Lin
// Date: 2022.11.28

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int prev_min = INT_MAX;
        int max_profit = INT_MIN;
        for (int price: prices){
            prev_min = min(price, prev_min);
            int cur_profit = price - prev_min;
            max_profit = max(cur_profit, max_profit);
        }
        return max_profit;
    }
};